exporters/etw/include/opentelemetry/exporters/etw/etw_data.h

```diff
@@ -0,0 +1,534 @@
+
+#include "opentelemetry/exporters/etw/etw_provider_exporter.h"
+
+#include "opentelemetry/exporters/etw/utils.h"
+
+namespace core  = opentelemetry::core;
+namespace trace = opentelemetry::trace;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+
+namespace exporter
+{
+namespace ETW
+{
+class Span;
+
+/// <summary>
+/// stream::Tracer class that allows to send spans to ETW
+/// </summary>
+class Tracer : public trace::Tracer
+{
+  /// <summary>
+  /// Parent provider of this Tracer
+  /// </summary>
+  trace::TracerProvider &parent;
+
+  /// <summary>
+  /// Provider Id (Name or GUID)
+  /// </summary>
+  std::string provId;
+
+  /// <summary>
+  /// Provider Handle
+  /// </summary>
+  ETWProvider::Handle provHandle;
+
+  /// <summary>
+  /// Encoding (Manifest, MessagePack or XML)
+  /// </summary>
+  ETWProvider::EventFormat encoding;
+
+  /// <summary>
+  /// ETWProvider is a singleton that aggregates all ETW writes.
+  /// </summary>
+  /// <returns></returns>
+  static ETWProvider &etwProvider()
+  {
+    static ETWProvider instance;  // C++11 magic static
+    return instance;
+  };
+
+public:
+  /// <summary>
+  /// Tracer constructor
+  /// </summary>
+  /// <param name="parent">Parent TraceProvider</param>
+  /// <param name="providerId">providerId</param>
+  /// <returns>Tracer instance</returns>
+  Tracer(trace::TracerProvider &parent,
+         nostd::string_view providerId     = "",
+         ETWProvider::EventFormat encoding = ETWProvider::EventFormat::ETW_MANIFEST)
+      : trace::Tracer(),
+        parent(parent),
+        provId(providerId.data(), providerId.size()),
+        encoding(encoding)
+  {
+    provHandle = etwProvider().open(provId, encoding);
+  };
+
+  /// <summary>
+  /// Start Span
+  /// </summary>
+  /// <param name="name">Span name</param>
+  /// <param name="options">Span options</param>
+  /// <returns>Span</returns>
+  virtual nostd::shared_ptr<trace::Span> StartSpan(
+      nostd::string_view name,
+      const common::KeyValueIterable &attributes,
+      const trace::SpanContextKeyValueIterable &links,
+      const trace::StartSpanOptions &options = {}) noexcept override
+  {
+    // TODO: support attributes
+    UNREFERENCED_PARAMETER(attributes);
+    UNREFERENCED_PARAMETER(links);
+    return trace::to_span_ptr<Span, Tracer>(this, name, options);
+  };
+
+  /// <summary>
+  /// Force flush data to Tracer, spending up to given amount of microseconds to flush.
+  /// </summary>
+  /// <param name="timeout">Allow Tracer to drop data if timeout is reached</param>
+  /// <returns>void</returns>
+  virtual void ForceFlushWithMicroseconds(uint64_t) noexcept override{};
+
+  /// <summary>
+  /// Close tracer, spending up to given amount of microseconds to flush and close.
+  /// </summary>
+  /// <param name="timeout">Allow Tracer to drop data if timeout is reached</param>
+  /// <returns></returns>
+  virtual void CloseWithMicroseconds(uint64_t) noexcept override
+  {
+    etwProvider().close(provHandle);
```

Comment:
Could you have, in one process, multiple tracers sharing the same provider handle? If so, would then closing one tracer also close the provider handles of the other tracers?